pkgs/servers/web-apps/snipe-it/default.nix

```diff
@@ -1,52 +1,52 @@
 { lib
-, pkgs
-, stdenv
-, fetchFromGitHub
 , dataDir ? "/var/lib/snipe-it"
+, fetchFromGitHub
 , mariadb
 , nixosTests
 , php
-, phpPackages
 }:
 
-let
-  package = (import ./composition.nix {
-    inherit pkgs php phpPackages;
-    inherit (stdenv.hostPlatform) system;
-    noDev = true; # Disable development dependencies
-  }).overrideAttrs (attrs : {
-    installPhase = attrs.installPhase + ''
-      # Before symlinking the following directories, copy the invalid_barcode.gif
-      # to a different location. The `snipe-it-setup` oneshot service will then
-      # copy the file back during bootstrap.
-      mkdir -p $out/share/snipe-it
-      cp $out/public/uploads/barcodes/invalid_barcode.gif $out/share/snipe-it/
-
-      rm -R $out/storage $out/public/uploads $out/bootstrap/cache
-      ln -s ${dataDir}/.env $out/.env
-      ln -s ${dataDir}/storage $out/
-      ln -s ${dataDir}/public/uploads $out/public/uploads
-      ln -s ${dataDir}/bootstrap/cache $out/bootstrap/cache
-
-      chmod +x $out/artisan
-
-      substituteInPlace config/database.php --replace "env('DB_DUMP_PATH', '/usr/local/bin')" "env('DB_DUMP_PATH', '${mariadb}/bin')"
-    '';
-  });
-
-in package.override rec {
+php.buildComposerProject (finalAttrs: {
   pname = "snipe-it";
   version = "6.2.2";
 
   src = fetchFromGitHub {
     owner = "snipe";
-    repo = pname;
-    rev = "v${version}";
-    sha256 = "11i9ijkl7am5k48y7r5k6nki2827cd7mw3dr1xj8dvb8diwaskqi";
+    repo = finalAttrs.pname;
```

Comment:
Please don't use this.

`repo = pname;` is nice for DRY but creates a binding that goes too far.

See further information about this here: https://github.com/nix-community/nixpkgs-lint/issues/21